packages/opentelemetry-core/test/node/environment.test.ts

```diff
@@ -0,0 +1,198 @@
+      process.env.FOO = String(NaN);
+      assert.strictEqual(getNumberFromEnv('FOO'), undefined);
+    });
+  });
+
+  describe('getStringListFromEnv', function () {
+    afterEach(function () {
+      delete process.env.FOO;
+    });
+
+    it('should treat empty string as undefined', function () {
+      process.env.FOO = '';
+      assert.strictEqual(getStringListFromEnv('FOO'), undefined);
+    });
+
+    it('should treat not defined as undefined', function () {
+      delete process.env.FOO;
+      assert.strictEqual(getStringListFromEnv('FOO'), undefined);
+    });
+
+    it('should trim extra whitespace', function () {
+      process.env.FOO = '  foo, bar,   ';
+      assert.deepStrictEqual(getStringListFromEnv('FOO'), ['foo', 'bar']);
+    });
+
+    it('should extract list from env var', function () {
+      process.env.FOO = 'foo,bar,baz';
+      assert.deepStrictEqual(getStringListFromEnv('FOO'), [
+        'foo',
+        'bar',
+        'baz',
+      ]);
+    });
+
+    it('should skip empty entries', function () {
+      process.env.FOO = '  ,undefined,,null, ,empty,';
+      assert.deepStrictEqual(getStringListFromEnv('FOO'), [
+        'undefined',
+        'null',
+        'empty',
+      ]);
+    });
+
+    it('should retain casing', function () {
+      process.env.FOO = 'fOo,BaR';
+      assert.deepStrictEqual(getStringListFromEnv('FOO'), ['fOo', 'BaR']);
+    });
+  });
+
+  describe('getBooleanFromEnv', function () {
+    afterEach(function () {
+      delete process.env.FOO;
+      sinon.restore();
+    });
+    it('should treat empty string as undefined', function () {
+      process.env.FOO = '';
+      assert.strictEqual(getBooleanFromEnv('FOO'), undefined);
+
+      process.env.FOO = '  ';
+      assert.strictEqual(getBooleanFromEnv('FOO'), undefined);
+    });
+
+    it('should treat not defined as undefined', function () {
+      delete process.env.FOO;
+      assert.strictEqual(getBooleanFromEnv('FOO'), undefined);
+    });
+
+    it('should trim extra whitespace', function () {
+      process.env.FOO = '  true';
+      assert.strictEqual(getBooleanFromEnv('FOO'), true);
+      process.env.FOO = 'false   ';
+      assert.strictEqual(getBooleanFromEnv('FOO'), false);
+      process.env.FOO = ' true ';
+      assert.strictEqual(getBooleanFromEnv('FOO'), true);
+    });
+
+    it('should ignore casing', function () {
+      process.env.FOO = 'tRUe';
+      assert.strictEqual(getBooleanFromEnv('FOO'), true);
+      process.env.FOO = 'TRUE';
+      assert.strictEqual(getBooleanFromEnv('FOO'), true);
+      process.env.FOO = 'FaLsE';
+      assert.strictEqual(getBooleanFromEnv('FOO'), false);
+    });
+
+    it('should ignore bogus data and warn', function () {
+      const warnStub = sinon.stub(diag, 'warn');
+      process.env.FOO = 'trueFALSE';
+      assert.strictEqual(getBooleanFromEnv('FOO'), undefined);
+      sinon.assert.calledOnceWithMatch(warnStub, 'Unknown value');
```

Comment:
nit: *could* also have asserts for diag.warn() being called from some of the other `get*FromEnv` functions above.

Reply:
also added in https://github.com/open-telemetry/opentelemetry-js/pull/5443/commits/ffe7e1e4389b5086bd4a4952d51b0ba4b6c0038e 🙂